balancer/catabalancer/sysstats.go

```diff
@@ -63,3 +97,52 @@ func GetSystemUsage() (SystemUsage, error) {
 
 	return systemUsage, nil
 }
+
+// Get bandwidth usage using the vnstat utility.
+// 'vnstat --json --iface en0 -tr 2.5' calculates traffic for given interface
+// over the specified duration in seconds
+func GetBandwidthUsage() (float64, error) {
+	iface := "eth0"
+	trafficArgs := []string{"-tr", "2.5"}
+	args := append([]string{"--json", "--iface", iface}, trafficArgs...)
+
+	// Run vnstat
+	cmd := exec.Command("vnstat", args...)
+	cmd.Env = os.Environ()
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return -1, err
+	}
+
+	// Parse json output
+	var data BandwidthData
+	err = json.Unmarshal([]byte(output), &data)
+	if err != nil {
+		return -1, err
+	}
+
+	// Get bandwidth in Mbits/s
+	totalBW := (float64(data.Rx.BytesPerSecond + data.Tx.BytesPerSecond)) / 8 / 1e6
+
+	// Get the NIC's speed
+	nicSpeedFilePath := fmt.Sprintf("/sys/class/net/%s/speed", iface)
+	file, err := os.Open(nicSpeedFilePath)
+	if err != nil {
+		return -1, err
+	}
+	defer file.Close()
+	content := make([]byte, 64)
+	n, err := file.Read(content)
+	if err != nil {
+		return -1, err
+	}
+	speedStr := strings.TrimSpace(string(content[:n]))
```

Comment:
it would be simpler and safer to just use `os.ReadFile()`

Reply:
Updated.